provider/pkg/provider/deployment_settings.go

```diff
@@ -0,0 +1,585 @@
+			}
+			if awsInput["policyARNs"].HasValue() && awsInput["policyARNs"].IsArray() {
+				policyARNsInput := awsInput["policyARNs"].ArrayValue()
+				policyARNs := make([]string, len(policyARNsInput))
+
+				for i, v := range policyARNsInput {
+					policyARNs[i] = getSecretOrStringValue(v)
+				}
+
+				aws.PolicyARNs = policyARNs
+			}
+
+			oidc.AWS = &aws
+		}
+
+		if oidcInput["gcp"].HasValue() && oidcInput["gcp"].IsObject() {
+			gcpInput := oidcInput["gcp"].ObjectValue()
+			var gcp pulumiapi.GCPOIDCConfiguration
+
+			if gcpInput["projectId"].HasValue() {
+				gcp.ProjectID = getSecretOrStringValue(gcpInput["projectId"])
+			}
+			if gcpInput["region"].HasValue() {
+				gcp.Region = getSecretOrStringValue(gcpInput["region"])
+			}
+			if gcpInput["workloadPoolId"].HasValue() {
+				gcp.WorkloadPoolID = getSecretOrStringValue(gcpInput["workloadPoolId"])
+			}
+			if gcpInput["providerId"].HasValue() {
+				gcp.ProviderID = getSecretOrStringValue(gcpInput["providerId"])
+			}
+			if gcpInput["serviceAccount"].HasValue() {
+				gcp.ServiceAccount = getSecretOrStringValue(gcpInput["serviceAccount"])
+			}
+			if gcpInput["tokenLifetime"].HasValue() {
+				gcp.TokenLifetime = getSecretOrStringValue(gcpInput["tokenLifetime"])
+			}
+
+			oidc.GCP = &gcp
+		}
+
+		if oidcInput["azure"].HasValue() && oidcInput["azure"].IsObject() {
+			azureInput := oidcInput["azure"].ObjectValue()
+			var azure pulumiapi.AzureOIDCConfiguration
+
+			if azureInput["tenantId"].HasValue() {
+				azure.TenantID = getSecretOrStringValue(azureInput["tenantId"])
+			}
+			if azureInput["clientId"].HasValue() {
+				azure.ClientID = getSecretOrStringValue(azureInput["clientId"])
+			}
+			if azureInput["subscriptionId"].HasValue() {
+				azure.SubscriptionID = getSecretOrStringValue(azureInput["subscriptionId"])
+			}
+
+			oidc.Azure = &azure
+		}
+
+		oc.OIDC = &oidc
+	}
+
+	return &oc
+}
+
+func getSecretOrStringValue(prop resource.PropertyValue) string {
+	switch prop.V.(type) {
+	case *resource.Secret:
+		return prop.SecretValue().Element.StringValue()
+	default:
+		return prop.StringValue()
+	}
+}
+
+func (ds *PulumiServiceDeploymentSettingsResource) Diff(req *pulumirpc.DiffRequest) (*pulumirpc.DiffResponse, error) {
+	olds, err := plugin.UnmarshalProperties(req.GetOlds(), plugin.MarshalOptions{KeepUnknowns: false, SkipNulls: true})
+	if err != nil {
+		return nil, err
+	}
+
+	news, err := plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{KeepUnknowns: true, SkipNulls: true})
+	if err != nil {
+		return nil, err
+	}
+
+	diffs := olds.Diff(news)
+	if diffs == nil {
+		return &pulumirpc.DiffResponse{
+			Changes: pulumirpc.DiffResponse_DIFF_NONE,
+		}, nil
+	}
+
+	replaces := make([]string, len(diffs.ChangedKeys()))
+	for i, k := range diffs.ChangedKeys() {
+		replaces[i] = string(k)
+	}
+
+	return &pulumirpc.DiffResponse{
+		Changes:             pulumirpc.DiffResponse_DIFF_SOME,
+		Replaces:            replaces,
+		DeleteBeforeReplace: true,
+	}, nil
```

Comment:
You can use [plugin.NewDetailedDiffFromObjectDiff](https://pkg.go.dev/github.com/pulumi/pulumi/sdk/v3@v3.61.1/go/common/resource/plugin#NewDetailedDiffFromObjectDiff) to convert `diff` to a detailed diff that you can return from this method. Personally I'd prefer that to using the deprecated `Replaces` field. You will need to post-process the result to change any diffs to replaces.

Reply:
Any advice on how to go from `plugin.PropertyDiff` to `pulumirpc.PropertyDiff`?

Reply:
nvm I think I got it

Reply:
Hm, are you saying you don't want me to use the `Replaces` field? It doesn't mention anywhere that it is deprecated and not using it turns changes into an `update` instead of a `replace` even if every field in the DetailedDiff is updated to have a `Kind` of `Replace`.

Reply:
> not using it turns changes into an update instead of a replace even if every field in the DetailedDiff is updated to have a Kind of Replace.

Huh, that's surprising. That feels like a bug.

Reply:
okay nvm again figured it out :)